mypy/nodes.py

```diff
@@ -171,10 +171,7 @@ class Node(Context):
     __slots__ = ()
 
     def __str__(self) -> str:
-        ans = self.accept(mypy.strconv.StrConv(options=Options()))
-        if ans is None:
-            return repr(self)
-        return ans
+        return self.accept(mypy.strconv.StrConv(options=Options()))
```

Comment:
I would not do this change. All node visitors can actually return `None` because of this exception https://github.com/python/mypy/blob/master/mypy_self_check.ini#L16-L18

Reply:
You can put a type ignore instead with a comment (node there is also relevant TODO in `NodeVisitor` docstring).

Reply:
> You can put a type ignore instead with a comment (note there is also relevant TODO in `NodeVisitor` docstring).

Is the TODO just saying to have a `raise NotImplementedException` in every body?

Reply:
@A5rocks as an option, yes. But it is probably still too risky. I think I tried it once couple years ago, and it caused a bunch of failures. Btw if you are curious you can try and see what happens with the tests, but this will allow to find some bugs that can be fixed incrementally.